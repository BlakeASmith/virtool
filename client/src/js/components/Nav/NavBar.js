/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */

import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Bar from "./Bar";

const mapStateToProps = (state) => {
    return state.account;
};

const mapDispatchToProps = () => {
    return {};
};

const NavBar = withRouter(connect(
    mapStateToProps,
    mapDispatchToProps
)(Bar));

export default NavBar;
