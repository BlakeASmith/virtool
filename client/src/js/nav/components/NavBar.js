import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { logout } from "../../account/actions";
import { DropdownItem, Icon, VTLogo } from "../../base";
import { getSoftwareUpdates } from "../../updates/actions";
import { isHomeActive } from "../utils";
import { NavDropdown } from "./Dropdown";
import { NavBarItem } from "./NavBarItem";
import Update from "./Update";

const NavBarLeft = styled.div`
    display: flex;
    align-items: center;
`;

const NavBarRight = styled.div`
    display: flex;
    align-items: center;
    margin-right: calc(100% - 100vw + 20px);
`;

const NavBarLogo = styled(VTLogo)`
    margin: 0 30px 0 35px;
`;

const StyledNavBar = styled.div`
    background-color: ${props => props.theme.color.primary};
    color: white;
    display: flex;
    height: 45px;
    justify-content: space-between;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 90;
`;

export class Bar extends React.Component {
    componentDidMount() {
        this.props.onGet();
    }

    render() {
        return (
            <StyledNavBar>
                <NavBarLeft>
                    <NavBarLogo />

                    <NavBarItem to="/home" isActive={isHomeActive}>
                        Home
                    </NavBarItem>

                    <NavBarItem to="/jobs">Jobs</NavBarItem>

                    <NavBarItem to="/samples">Samples</NavBarItem>

                    <NavBarItem to="/refs">References</NavBarItem>

                    <NavBarItem to="/hmm">HMM</NavBarItem>

                    <NavBarItem to="/subtraction">Subtraction</NavBarItem>
                </NavBarLeft>

                <NavBarRight>
                    <Update />

                    <NavBarItem target="_blank" to="//gitter.im/virtool/virtool" rel="noopener noreferrer">
                        <Icon name="comments" />
                    </NavBarItem>

                    <NavBarItem
                        target="_blank"
                        to="//virtool.ca/docs/manual"
                        rel="noopener noreferrer"
                        style={{ paddingLeft: 0 }}
                    >
                        <Icon name="book" />
                    </NavBarItem>

                    <NavDropdown userId={this.props.id}>
                        <DropdownItem to="/account">Account</DropdownItem>

                        {this.props.administrator && <DropdownItem to="/administration">Administration </DropdownItem>}

                        <DropdownItem
                            to="//gitreports.com/issue/virtool/virtool"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            Report Issue
                        </DropdownItem>

                        <DropdownItem onClick={this.props.logout}>Logout</DropdownItem>
                    </NavDropdown>
                </NavBarRight>
            </StyledNavBar>
        );
    }
}

export const mapStateToProps = state => ({
    ...state.account,
    pending: state.app.pending
});

export const mapDispatchToProps = dispatch => ({
    logout: () => {
        dispatch(logout());
    },

    onGet: () => {
        dispatch(getSoftwareUpdates());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Bar);
