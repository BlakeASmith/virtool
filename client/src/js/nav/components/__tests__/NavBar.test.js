import { Bar, mapStateToProps, mapDispatchToProps } from "../NavBar";

describe("<Bar />", () => {
    const props = {
        onGet: jest.fn(),
        id: "foo",
        administrator: true,
        logout: jest.fn(),
        pending: false
    };
    it("should render", () => {
        const wrapper = shallow(<Bar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps", () => {
    const state = {
        account: { foo: "bar" },
        app: {
            pending: false
        }
    };
    it("should return props", () => {
        const result = mapStateToProps(state);
        expect(result).toEqual({
            foo: "bar",
            pending: false
        });
    });
});

describe("mapDispatchToProps", () => {
    const dispatch = jest.fn();

    it("should return logout in props", () => {
        const props = mapDispatchToProps(dispatch);
        props.logout();
        expect(dispatch).toHaveBeenCalled();
    });

    it("should return onGet in props", () => {
        const props = mapDispatchToProps(dispatch);
        props.onGet();
        expect(dispatch).toHaveBeenCalled();
    });
});
