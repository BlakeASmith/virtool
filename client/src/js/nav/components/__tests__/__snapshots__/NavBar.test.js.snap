// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> should render 1`] = `
<NavBar__StyledNavBar>
  <NavBar__NavBarLeft>
    <NavBar__NavBarLogo />
    <NavBarItem
      isActive={[Function]}
      to="/home"
    >
      Home
    </NavBarItem>
    <NavBarItem
      to="/jobs"
    >
      Jobs
    </NavBarItem>
    <NavBarItem
      to="/samples"
    >
      Samples
    </NavBarItem>
    <NavBarItem
      to="/refs"
    >
      References
    </NavBarItem>
    <NavBarItem
      to="/hmm"
    >
      HMM
    </NavBarItem>
    <NavBarItem
      to="/subtraction"
    >
      Subtraction
    </NavBarItem>
  </NavBar__NavBarLeft>
  <NavBar__NavBarRight>
    <Connect(NotificationIcon) />
    <NavBarItem
      rel="noopener noreferrer"
      target="_blank"
      to="//gitter.im/virtool/virtool"
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="comments"
      />
    </NavBarItem>
    <NavBarItem
      rel="noopener noreferrer"
      style={
        Object {
          "paddingLeft": 0,
        }
      }
      target="_blank"
      to="//virtool.ca/docs/manual"
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="book"
      />
    </NavBarItem>
    <NavDropdown
      userId="foo"
    >
      <DropdownItem
        to="/account"
      >
        Account
      </DropdownItem>
      <DropdownItem
        to="/administration"
      >
        Administration 
      </DropdownItem>
      <DropdownItem
        rel="noopener noreferrer"
        target="_blank"
        to="//gitreports.com/issue/virtool/virtool"
      >
        Report Issue
      </DropdownItem>
      <DropdownItem
        onClick={[MockFunction]}
      >
        Logout
      </DropdownItem>
    </NavDropdown>
  </NavBar__NavBarRight>
</NavBar__StyledNavBar>
`;
